### PR TITLE
Build plottable.min.js with webpack

### LINKED ISSUE #3315: plottable.min.js doesn't have Copyright header
https://unpkg.com/plottable@3.0.0-beta.4/plottable.min.js

### PR DESCRIPTION
**Pull request by hellochar**
Use webpack to build plottable.min.js, removing old grunt code and opening the door for more enhancements later (sourcemaps, dev logging that gets wiped in prod, removing comments)

Fixes #3315

### CODE REVIEW

File: webpackConfig/build-min.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+const UglifyJSPlugin = require('uglifyjs-webpack-plugin');
```
Comment (adidahiya) [inline webpackConfig/build-min.js]: let's rename this to `config/webpack.config.prod.js`, it's more consistent with other projects

File: webpackConfig/build-min.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+const webpackMerge = require("webpack-merge");
```
Comment (adidahiya) [inline webpackConfig/build-min.js]: is this to do deep merging of objects? I don't love the idea of pulling in another random dependency for this

File: webpackConfig/build.js [hunk 3]
```diff
@@ -0,0 +1,79 @@
+const webpack = require("webpack");
```
Comment (adidahiya) [inline webpackConfig/build.js:1]: this is the base config right? let's call it `config/webpack.config.js`

File: webpackConfig/build.js [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+const WriteFilePlugin = require("write-file-webpack-plugin");
```
Comment (adidahiya) [inline webpackConfig/build.js]: why do you need this? are entry points + output not sufficient?

File: webpackConfig/test.js [hunk 5]
```diff
@@ -0,0 +1,39 @@
+const webpack = require("webpack");
```
Comment (adidahiya) [inline webpackConfig/test.js:1]: `config/webpack.config.test.js`

File: package.json [hunk 6]
```diff
@@ -53,8 +50,11 @@
     "sinon": "^2.1.0",
     "tslint": "4.5.1",
     "typescript": "2.1.5",
+    "uglify-js": "^2.8.18",
+    "uglifyjs-webpack-plugin": "^0.3.1",
```
Comment (adidahiya) [inline package.json]: webpack includes minification out of the box... are you sure you need this?

**Review by adidahiya — DISMISSED** [review]

File: webpackConfig/build.js (hunk 3 above)
Comment (hellochar) [reply webpackConfig/build.js:1]: The test config is totally separate, and we have a real `/webpack.config.js` that runs both `test` and `build` - see https://github.com/palantir/plottable/pull/3318/files#diff-11e9f7f953edc64ba14b0cc350ae7b9dR5 for why

File: webpackConfig/build-min.js (hunk 2 above)
Comment (hellochar) [reply webpackConfig/build-min.js]: They use it throughout the webpack docs: https://webpack.js.org/guides/production-build/#the-manual-way-configuring-webpack-for-multiple-environments and it has features to do "smart" merging of specific webpack properties which might be useful in the future

Comment (adidahiya) [reply webpackConfig/build-min.js]: alright fair enough

File: webpackConfig/build.js (hunk 4 above)
Comment (hellochar) [reply webpackConfig/build.js]: this wrote plottable.js to disk but is unneeded now, good catch

**Review by adidahiya — APPROVED** [review]

**Pull request merged**